pkg/destroy/vsphere/vsphere.go

```diff
@@ -33,8 +33,9 @@ type ClusterUninstaller struct {
 	ClusterID string
 	InfraID   string
 
-	Client     *vim25.Client
-	RestClient *rest.Client
+	Client       *vim25.Client
+	RestClient   *rest.Client
+	ClientLogout vsphere.ClientLogout
```

Comment:
This seems like a smell that `New` is causing resources to be held rather than having all resource acquisition and cleanup contained within `Run`. It is particularly smelly that this leaks into the interface affecting the destroyers of other platforms, too.

Reply:
Yes, I see how this could be solved all inside `Run` as well, without adding the `Cleanup`. I'm happy to go that way if that looks better to you.

Reply:
> This seems like a smell that New is causing resources to be held rather than having all resource acquisition and cleanup contained within Run

It looks like there's no particular pattern with which other destroyers are managing their API client. 

For instance: AWS, alibaba, Azure, and ibmcloud all keep a session or entire client inside of their `ClusterUninstaller` struct. On the other hand, oVirt manages the API client entirely within `Run`. 

> It is particularly smelly that this leaks into the interface affecting the destroyers of other platforms, too.

If I understand what you are saying correctly, I can explain that the new `Cleanup` would promote a separation of concerns for the Destroyer. vSphere isn't the only one that would benefit from this. Alibaba, oVirt, and baremetal (not sure this is used) could all take advantage of the new `Cleanup`. 

I'm curious what you think about this or if you still think consolidating into just `Run` is still the best move.